### DISCUSSION TITLE
kOps 1.26.x version soon?

### DISCUSSION THREAD

**Discussion by argoyle** [eddycharly/terraform-provider-kops#948] [General]
Hi @eddycharly !

Any plans for a new release soon?

**Comment by eddycharly** [comment]
Hi, I can try to cut one soon :)

**Reply by eddycharly** [reply]
I don't have a lot of time lately, thanks for pinging me !

**Reply by argoyle** [reply]
No worries, just shout if there is anything I can help out with.

**Reply by argoyle** [reply]
Took a look at the necessary changes today and it feels like there are quite a lot of changes to the kOps API (v1alpha2 -> v1alpha3). I currently have something which crashes during docs templating 😂 💥

**Reply by eddycharly** [reply]
> I currently have something which crashes during docs templating

It's a good start 😂

**Reply by argoyle** [reply]
I might have some free time the following evenings, will let you know how/if I make any progress.

**Reply by eddycharly** [reply]
Made some progress https://github.com/eddycharly/terraform-provider-kops/pull/958

**Reply by eddycharly** [reply]
Migration of existing clusters is not going to be easy though

**Reply by eddycharly** [reply]
I will probably release an alpha version today... not sure how to handle upgrading the provider though.

**Reply by argoyle** [reply]
Might be easier to recommend a `terraform state rm <resource>` followed by a `terraform import <resource>` for this upgrade?

**Reply by eddycharly** [reply]
Yeah, will recommend that if no other options... will ask the kops maintainers if there's something we can do to easily convert from alpha2 to alpha3 🤷

**Reply by argoyle** [reply]
Think I saw some conversion-func in the commits when checking earlier

**Reply by eddycharly** [reply]
[v1.26.0-alpha.1](https://github.com/eddycharly/terraform-provider-kops/tree/v1.26.0-alpha.1) has been cut, let me know if you get a chance to try it out 🤞

**Reply by argoyle** [reply]
Nice! Will test soon.

**Reply by argoyle** [reply]
Hmmm, seems like the Terraform Registry has not been updated?

**Reply by eddycharly** [reply]
😅 should be good now, i forced a resync 🤷

**Reply by argoyle** [reply]
Seems better now. Just got a validate and a plan through for a new cluster and the plan seems solid. Will try to apply it later today or tomorrow.

**Reply by eddycharly** [reply]
@argoyle did you get a chance to try applying ?

**Reply by argoyle** [reply]
Sorry, weekend happened. 😂  Will try now.

**Reply by eddycharly** [reply]
🙈

**Reply by argoyle** [reply]
Apply worked fine. ❤️  Time to destroy again, create using previous version and then try some kind of upgrade path.

**Reply by argoyle** [reply]
Hmmm, why would I get this when trying the previous version all of a sudden? I've only switched back to my unmodified module and the provider version. Could there be some left overs somewhere?

```
╷
│ Error: kops upgrade is required
│
│   with module.k8s.kops_cluster_updater.k8s_updater,
│   on .terraform/modules/k8s/k8s.tf line 339, in resource "kops_cluster_updater" "k8s_updater":
│  339: resource "kops_cluster_updater" "k8s_updater" {
│
```

**Reply by eddycharly** [reply]
Could be related to this change https://github.com/eddycharly/terraform-provider-kops/pull/980

**Reply by argoyle** [reply]
Hmmm, but that is not in 1.25.3, right?

**Reply by eddycharly** [reply]
Looks like you can use `KOPS_RUN_OBSOLETE_VERSION` env var to bypass it

**Reply by argoyle** [reply]
Ok, trying that

**Reply by eddycharly** [reply]
Seems to be there in kops https://github.com/kubernetes/kops/blob/e299cc2c4b90c451d3d868da463c965646bacb80/upup/pkg/fi/cloudup/apply_cluster.go#L898

**Reply by eddycharly** [reply]
@argoyle any news ?

**Reply by argoyle** [reply]
No, sorry, time has been sparse this week. Probably won't have time until next week.

**Reply by kuldazbraslav** [reply]
Is there a way to define [stateUpgraders](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/state-migration) manually? From what I could see, the provider only uses a generic upgrader: https://github.com/eddycharly/terraform-provider-kops/blob/main/hack/gen-tf-code/templates/schema.tpl#L98

**Reply by eddycharly** [reply]
FYI i'm trying to switch to `v1alpha2` instead of internal api https://github.com/eddycharly/terraform-provider-kops/pull/1057
It will probably be a breaking change but I hope it will make the provider more stable in the future 🤞

**Reply by argoyle** [reply]
Finally got back to this. Created a new cluster using 1.25.4 and upgrade to 1.26.0-alpha.1 actually seems to work. Still running but it has replaced one master at least.

**Reply by argoyle** [reply]
And now all masters and nodes have been replaced. Worked without any issues. ❤️

**Reply by argoyle** [reply]
No need for me at least to rm/import anything

**Reply by argoyle** [reply]
Long time no hear! Now that K8S 1.28 is out as well it feels like we would need some traction on this one (and the next few versions). I assume that you're still lacking time @eddycharly ? Anything we can do to help out?
